src/main/java/org/wso2/carbon/module/swiftiso20022/utils/MT940ValidationUtils.java

```diff
@@ -149,35 +150,35 @@ public static ErrorModel validateMT940Format(Map<String, Object> fields) {
      * @param referenceValue  Reference to be validated
      * @return    Whether reference is valid
      */
-    public static ErrorModel validateReference(String referenceValue, String fieldName) {
+    public static ValidationResult validateReference(String referenceValue, String fieldName) {
         String[] referenceDetails = referenceValue.split(ConnectorConstants.COLON);
         if (!isListLengthValid(referenceDetails)) {
-            return new ErrorModel(ConnectorConstants.INVALID_REQUEST_PAYLOAD,
+            return new ValidationResult(ConnectorConstants.INVALID_REQUEST_PAYLOAD,
                     String.format(ConnectorConstants.ERROR_PARAMETER_INVALID, fieldName));
         }
 
         String reference = referenceDetails[2];
         if (StringUtils.isBlank(reference)) {
-            return new ErrorModel(ConnectorConstants.INVALID_REQUEST_PAYLOAD,
+            return new ValidationResult(ConnectorConstants.INVALID_REQUEST_PAYLOAD,
                     String.format(ConnectorConstants.ERROR_PARAMETER_MISSING, fieldName));
         }
 
         if (reference.length() > 16) {
-            return new ErrorModel(ConnectorConstants.ERROR_M50,
+            return new ValidationResult(ConnectorConstants.ERROR_M50,
                     String.format(ConnectorConstants.ERROR_PARAMETER_LENGTH, fieldName, 16));
```

Comment:
use the constant here as well and check other places

Reply:
Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/6/commits/02976704c3798e54f6e5b27d539a212b8bdb900d